core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataUpdateSettingsService.java

```diff
@@ -267,57 +221,67 @@ public ClusterState execute(ClusterState currentState) {
                 }
 
                 ClusterBlocks.Builder blocks = ClusterBlocks.builder().blocks(currentState.blocks());
-                Boolean updatedReadOnly = openSettings.getAsBoolean(IndexMetaData.SETTING_READ_ONLY, null);
-                if (updatedReadOnly != null) {
-                    for (String index : actualIndices) {
-                        if (updatedReadOnly) {
-                            blocks.addIndexBlock(index, IndexMetaData.INDEX_READ_ONLY_BLOCK);
-                        } else {
-                            blocks.removeIndexBlock(index, IndexMetaData.INDEX_READ_ONLY_BLOCK);
-                        }
+                final boolean updatedReadOnly = IndexMetaData.INDEX_READ_ONLY_SETTING.get(openSettings);
```

Comment:
If looks like if you didn't specify the value it used to leave it unchanged on all the indexes but now it'll remove all the blocks I think.